services/export/shared/transactionsExport.js

```diff
@@ -0,0 +1,462 @@
+
+	const interval = await standardizeIntervalFromParams(params);
+	const [from, to] = interval.split(':');
+	const range = moment.range(moment(from, DATE_FORMAT), moment(to, DATE_FORMAT));
+	const arrayOfDates = (Array.from(range.by('day'))).map(d => d.format(DATE_FORMAT));
+
+	for (let i = 0; i < arrayOfDates.length; i++) {
+		/* eslint-disable no-await-in-loop */
+		const day = arrayOfDates[i];
+		const partialFilename = await getPartialFilenameFromParams(params, day);
+		if (await partials.exists(partialFilename)) {
+			const transactions = JSON.parse(await partials.read(partialFilename));
+			allTransactions.push(...transactions);
+		} else if (await noTransactionsCache.get(partialFilename) !== true) {
+			const fromTimestampPast = moment(day, DATE_FORMAT).startOf('day').unix();
+			const toTimestampPast = moment(day, DATE_FORMAT).endOf('day').unix();
+			const timestamp = `${fromTimestampPast}:${toTimestampPast}`;
+			const { data: transactions } = await requestAll(
+				getTransactionsInAsc, { ...params, timestamp },
+				MAX_NUM_TRANSACTIONS,
+			);
+			allTransactions.push(...transactions);
+
+			const incomingCrossChainTransferTxs = await getCrossChainTransferTransactionInfo({
+				...params,
+				timestamp,
+			});
+
+			if (incomingCrossChainTransferTxs.length) {
+				allTransactions.push(...incomingCrossChainTransferTxs);
+			}
+
+			if (day !== getToday()) {
+				if (transactions.length) {
+					partials.write(partialFilename, JSON.stringify(transactions));
+				} else {
+					// Flag to prevent unnecessary calls to core/storage space usage on the file cache
+					const RETENTION_PERIOD = getDaysInMilliseconds(config.cache.partials.retentionInDays);
+					await noTransactionsCache.set(partialFilename, true, RETENTION_PERIOD);
+				}
+			}
+		}
+		/* eslint-enable no-await-in-loop */
+	}
+
+	const excelFilename = await getExcelFilenameFromParams(params);
+	const currentChainID = await getCurrentChainID();
+
+	// Add duplicate entry with zero fees for self token transfer transactions
+	allTransactions.forEach((tx, i, arr) => {
+		if (checkIfSelfTokenTransfer(tx) && !tx.isSelfTokenTransferCredit) {
+			arr.splice(i + 1, 0, { ...tx, fee: '0', isSelfTokenTransferCredit: true });
+		}
+	});
+
+	const normalizedTransactions = await Promise.all(allTransactions
+		.map(async (t) => normalizeTransaction(getAddressFromParams(params), t, currentChainID)));
+	const metadata = await getMetadata(params, currentChainID);
+
+	const workBook = new excelJS.Workbook();
+	const transactionExportSheet = workBook.addWorksheet(config.excel.sheets.TRANSACTION_HISTORY);
+	const metadataSheet = workBook.addWorksheet(config.excel.sheets.METADATA);
+	transactionExportSheet.columns = fields.transactionMappings;
+	metadataSheet.columns = fields.metadataMappings;
+
+	transactionExportSheet.addRows(normalizedTransactions);
+	metadataSheet.addRows([metadata]);
+
+	await workBook.xlsx.writeFile(`${config.cache.exports.dirPath}/${excelFilename}`);
```

Comment:
Please check above.

Also, please double check if we handle the `accountInitializationFee` for crosssChainTransfers.